src/v0/destinations/braze/braze.util.test.js

```diff
@@ -0,0 +1,725 @@
+        { message: { anonymousId: 'anon456' } },
+      ];
+      const expectedOutput = {
+        externalIdsToQuery: [],
+        aliasIdsToQuery: ['anon123', 'anon456'],
+      };
+      const actualOutput = BrazeDedupUtility.prepareInputForDedup(input);
+      expect(actualOutput).toEqual(expectedOutput);
+    });
+  });
+
+  describe('prepareChunksForDedup', () => {
+    it('should return an empty array if both externalIdsToQuery and aliasIdsToQuery are empty', () => {
+      const externalIdsToQuery = [];
+      const aliasIdsToQuery = [];
+      const expectedOutput = [];
+      const actualOutput = BrazeDedupUtility.prepareChunksForDedup(
+        externalIdsToQuery,
+        aliasIdsToQuery,
+      );
+      expect(actualOutput).toEqual(expectedOutput);
+    });
+
+    it('should return an array of chunks with external_id identifiers if externalIdsToQuery is not empty', () => {
+      const externalIdsToQuery = ['123', '456', '789'];
+      const aliasIdsToQuery = [];
+      const expectedOutput = [
+        [{ external_id: '123' }, { external_id: '456' }, { external_id: '789' }],
+      ];
+      const actualOutput = BrazeDedupUtility.prepareChunksForDedup(
+        externalIdsToQuery,
+        aliasIdsToQuery,
+      );
+      expect(actualOutput).toEqual(expectedOutput);
+    });
+
+    it('should return an array of chunks with alias_name and alias_label identifiers if aliasIdsToQuery is not empty', () => {
+      const externalIdsToQuery = [];
+      const aliasIdsToQuery = ['abc', 'def', 'ghi'];
+      const expectedOutput = [
+        [
+          { alias_name: 'abc', alias_label: 'rudder_id' },
+          { alias_name: 'def', alias_label: 'rudder_id' },
+          { alias_name: 'ghi', alias_label: 'rudder_id' },
+        ],
+      ];
+      const actualOutput = BrazeDedupUtility.prepareChunksForDedup(
+        externalIdsToQuery,
+        aliasIdsToQuery,
+      );
+      expect(actualOutput).toEqual(expectedOutput);
+    });
+
+    it('should return an array of chunks with both external_id and alias_name/alias_label identifiers if both arrays are not empty', () => {
+      const externalIdsToQuery = ['123', '456'];
+      const aliasIdsToQuery = ['abc', 'def', 'ghi'];
+      const expectedOutput = [
+        [
+          { external_id: '123' },
+          { external_id: '456' },
+          { alias_name: 'abc', alias_label: 'rudder_id' },
+          { alias_name: 'def', alias_label: 'rudder_id' },
+          { alias_name: 'ghi', alias_label: 'rudder_id' },
+        ],
+      ];
+      const actualOutput = BrazeDedupUtility.prepareChunksForDedup(
+        externalIdsToQuery,
+        aliasIdsToQuery,
+      );
+      expect(actualOutput).toEqual(expectedOutput);
+    });
+
+    it('should chunk the identifiers array into arrays of size 50', () => {
+      const externalIdsToQuery = new Array(100).fill('123');
+      const aliasIdsToQuery = new Array(150).fill('abc');
+      const expectedOutput = [
```

Comment:
shouldn't we take care of duplicate ids?

Reply:
That is covered in `prepareInputForDedup` function. The responsibility of `prepareChunksForDedup` is to chunk not to dedup